src/cljc/jinteki/decks.cljc

```diff
@@ -273,97 +228,104 @@
       :reason (join "\n" (filter identity (vals reasons)))
       :description description})))
 
-(defn onesies-legal
-  "Returns true if deck is valid under 1.1.1.1 format rules. https://www.reddit.com/r/Netrunner/comments/5238a4/1111_onesies/"
-  [sets deck]
-  (let [over-one-core (cards-over-one-core deck)
-        valid-sets ["Revised Core Set"]
-        restricted-sets (group-cards-from-restricted-sets sets valid-sets deck)
-        restricted-bigboxes (rest (:bigboxes restricted-sets)) ; one big box is fine
-        restricted-datapacks (rest (:datapacks restricted-sets)) ; one datapack is fine
-        only-one-offence (>= 1 (apply + (map count [over-one-core restricted-bigboxes restricted-datapacks]))) ; one offence is fine
-        example-card (fn [cardlist] (join ", " (map #(get-in % [:card :title]) (take 2 cardlist))))
-        reasons (if only-one-offence
-                  {}
-                  {:onecore (when (not= (count over-one-core) 0)
-                              (str "Only one Revised Core Set permitted - check: " (example-card over-one-core)))
-                   :bigbox (when (not= (count restricted-bigboxes) 0)
-                             (str "Only one Deluxe Expansion permitted - check: " (example-card (second (first restricted-bigboxes)))))
-                   :datapack (when (not= (count restricted-datapacks) 0)
-                               (str "Only one Datapack permitted - check: " (example-card (second (first restricted-datapacks)))))})]
-    {:legal (not-any? val reasons)
-     :reason (join "\n" (filter identity (vals reasons)))
-     :description "1.1.1.1 format compliant"}))
-
 ;; Card and deck validity
 (defn allowed?
   "Checks if a card is allowed in deck of a given identity - not accounting for influence"
-  [card {:keys [side faction code] :as identity}]
+  [card {:keys [side faction title] :as identity}]
   (and (not= (:type card) "Identity")
        (= (:side card) side)
        (or (not= (:type card) "Agenda")
            (= (:faction card) "Neutral")
            (= (:faction card) faction)
-           (is-draft-id? identity))
-       (or (not= code "03002") ; Custom Biotics: Engineered for Success
+           (draft-id? identity))
+       (or (not= title "Custom Biotics: Engineered for Success")
            (not= (:faction card) "Jinteki"))))
 
-(defn valid-deck? [{:keys [identity cards] :as deck}]
-  (and (not (nil? identity))
-       (>= (card-count cards) (min-deck-size identity))
-       (<= (influence-count deck) (id-inf-limit identity))
-       (every? #(and (allowed? (:card %) identity)
-                     (legal-num-copies? identity %)) cards)
-       (or (= (:side identity) "Runner")
-           (let [min (min-agenda-points deck)]
-             (<= min (agenda-points deck) (inc min))))))
+(defn check-deck-status
+  "Checks the valid and standard keys of a deck-status map to check if the deck is legal, casual or invalid."
+  [{:keys [valid standard eternal core-experience snapshot]}]
```

Comment:
We could make this less hard-coded I think. Just loop over keys and return the string version of the key if legal.

Reply:
Although that would lose internal order, so maybe this is the best way.

Reply:
Yeah, I want to return the smallest version first. Until we let players choose a deck's format and then only check it, we have to work from smallest to biggest available card pool because by definition all Core Experience decks are Standard legal too.

Reply:
Well, shortly players _will_ be able to choose a deck's format.